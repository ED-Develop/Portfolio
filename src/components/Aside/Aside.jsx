import React from 'react';
import style from './Aside.module.css';
import Nav from "./Nav/Nav";
import ProfileCard from "./ProfileCard/ProfileCard";
import TopFriends from "./TopFriends/TopFriends";

const Aside = ({avatar, login, myId, isAuth, friendsData}) => {
        return (
            <aside className={style.aside}>
                {isAuth && <ProfileCard avatar={avatar} login={login}/>}
                <Nav myId={myId}/>
                {isAuth && <TopFriends friendsData={friendsData}/>}
            </aside>
        )
};

export default Aside;