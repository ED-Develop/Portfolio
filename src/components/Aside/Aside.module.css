.aside {
    padding: 0 10px 0 8px;
}
