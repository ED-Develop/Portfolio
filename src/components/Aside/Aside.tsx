import React, {FC} from 'react';
import style from './Aside.module.css';
import Nav from "./Nav/Nav";
import {Affix, Layout} from "antd";

const {Sider} = Layout;

type PropsType = {
    isAuth: boolean,
    collapsed: boolean
}

const Aside: FC<PropsType> = ({isAuth, collapsed}) => {
        return (
            <Sider theme='light' width='20.8%' className={style.aside} collapsed={collapsed} collapsedWidth={'16.6%'}>
                <Affix offsetTop={65}>
                    <Nav collapsed={collapsed}/>
                </Affix>
            </Sider>
        )
};

export default Aside;
