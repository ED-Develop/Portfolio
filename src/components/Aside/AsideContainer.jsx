import React from 'react';
import Aside from "./Aside";
import {connect} from "react-redux";

let mapStateToProps=(state) => {
    return {
        friendsData: state.asidePage.friendsData,
        isAuth: state.auth.isAuth,
        login: state.auth.login,
        myId: state.auth.userId,
        avatar: state.auth.photos.small
    };
};

const AsideContainer = connect(mapStateToProps)(Aside);

export default AsideContainer;