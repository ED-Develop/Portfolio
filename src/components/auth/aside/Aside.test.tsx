import React from 'react';
import {render} from '@testing-library/react';
import {LoginAside} from './Aside';

describe('aside component: ', () => {
    test('Should renders correctly', () => {
        const {container} = render(<LoginAside/>);

        expect(container).toMatchSnapshot();
    })
})