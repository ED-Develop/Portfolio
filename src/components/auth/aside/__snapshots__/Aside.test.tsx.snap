// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside component:  Should renders correctly 1`] = `
<div>
  <aside
    class="sider ant-layout-sider ant-layout-sider-dark"
    style="flex: 0 0 33%; max-width: 33%; min-width: 33%; width: 33%;"
  >
    <div
      class="ant-layout-sider-children"
    >
      <div
        class="landing__sider"
      />
    </div>
  </aside>
</div>
`;

exports[`aside component:  Should renders correctly 1`] = `
<div>
  <aside
    class="sider ant-layout-sider ant-layout-sider-dark"
    style="flex: 0 0 33%; max-width: 33%; min-width: 33%; width: 33%;"
  >
    <div
      class="ant-layout-sider-children"
    >
      <div
        class="landing__sider"
      />
    </div>
  </aside>
</div>
`;
